src/data/pnat/basic.lean

```diff
@@ -349,4 +350,122 @@ theorem gcd_mul_lcm (n m : ℕ+) : (gcd n m) * (lcm n m) = n * m :=
 
 def prime (p : ℕ+) : Prop := (p : ℕ).prime
 
+/-- Two pnats are coprime if their gcd is 1. -/
+def coprime (m n : ℕ+) : Prop := m.gcd n = 1
+
+@[simp]
+lemma eq_iff {m n : ℕ+} :
+(m : ℕ) = ↑n ↔ m = n := by { split, apply eq, intro h, rw h }
+
+@[simp]
+lemma coprime_coe {m n : ℕ+} : nat.coprime ↑m ↑n ↔ m.coprime n :=
+by { unfold coprime, unfold nat.coprime, rw ← eq_iff, simp }
+
+lemma coprime.mul {k m n : ℕ+} : m.coprime k → n.coprime k → (m * n).coprime k :=
+by { repeat {rw ← coprime_coe}, rw mul_coe, apply nat.coprime.mul }
+
+lemma gcd_eq_left_iff_dvd {m n : ℕ+} : m ∣ n ↔ m.gcd n = m :=
+by { rw dvd_iff, rw nat.gcd_eq_left_iff_dvd, rw ← eq_iff, simp }
+
+lemma coprime.gcd_mul_right_cancel (m : ℕ+) {n k : ℕ+} :
+  k.coprime n → (m * k).gcd n = m.gcd n :=
+begin
+  intro h, apply eq, simp only [gcd_coe, mul_coe],
+  apply nat.coprime.gcd_mul_right_cancel, simpa
+end
+
+lemma gcd_comm {m n : ℕ+} : m.gcd n = n.gcd m :=
+by { apply eq, simp only [gcd_coe], apply nat.gcd_comm }
+
+@[simp]
+lemma one_gcd {n : ℕ+} : gcd 1 n = 1 :=
+by { rw ← gcd_eq_left_iff_dvd, apply one_dvd }
+
+@[simp]
+lemma gcd_one {n : ℕ+} : gcd n 1 = 1 := by { rw gcd_comm, apply one_gcd }
+
+lemma coprime.symm {m n : ℕ+} : m.coprime n → n.coprime m :=
+by { unfold coprime, rw gcd_comm, simp }
+
+@[simp]
+lemma one_coprime {n : ℕ+} : (1 : ℕ+).coprime n := one_gcd
+
+@[simp]
+lemma coprime_one {n : ℕ+} : n.coprime 1 := coprime.symm one_coprime
+
+lemma coprime.coprime_dvd_left {m k n : ℕ+} :
+  m ∣ k → k.coprime n → m.coprime n :=
+by { rw dvd_iff, repeat {rw ← coprime_coe}, apply nat.coprime.coprime_dvd_left }
+
+lemma coprime_factor_eq_gcd_left {a b m n : ℕ+} (cop : m.coprime n) (am : a ∣ m) (bn : b ∣ n):
```

Comment:
This one seems pretty specific.
However, if you add the symmetric version as I mentioned above, and you put it after the lemma `coprime.gcd_mul` just below it, then I think you can compress the proof so far that it's maybe not ever really worth stating this explicitly.
As a guide, I would say: prove whatever's done for `nat.coprime` and `nat.gcd`. Those APIs are quite complete.

Reply:
I've multiplied this one by 4 now. I hesitate to prove literally all of that API for pnats, when I think we might be able to develop a typeclass that encapsulates all the basic ideas, and could work for any gcd domain, and the associates in any monoid.